resources/process-autogenerated-form/src/js/app.js

```diff
@@ -83,14 +83,36 @@
         $scope.postData = function postData() {
             $scope.message = undefined;
             var jsonifiedDataToSend = jsonify($scope.dataToSend);
-            contractSrvc.startProcess(processId, jsonifiedDataToSend).then(function () {
-
-                $window.top.location.href = '/bonita';
-            }, function (reason) {
-                $scope.message = reason.data.explanations ? reason.data.explanations : reason.data.message;
-            });
+            contractSrvc.startProcess(processId, jsonifiedDataToSend).then(
+                onPostSuccess, onPostError);
         };
 
+        var onPostSuccess = function(response) {
+            //if the form is displayed in an iframe
+            if ($window.parent !== $window.self) {
+                notifyParentFrame('success', response.status);
+            } else {
```

Comment:
What happens if the parent does not want to handle the post message?
Should we redirect the current frame no matters what? The behaviour would be symetrical with what we do in case of error.